test_engine/terraform/aws/ubuntu/main.tf

```diff
@@ -0,0 +1,321 @@
+  }
+}
+
+# Create internet gateway
+resource "aws_internet_gateway" "build_engine_aws_igw" {
+  vpc_id = aws_vpc.build_engine_aws_vpc.id
+
+  tags = {
+    Name = "build_engine_igw"
+  }
+}
+
+# Create controlplane security group
+resource "aws_security_group" "build_engine_aws_secgrp_controlplane" {
+  name        = "build_engine_aws_secgrp_controlplane"
+  description = "Allow all inbound traffic"
+  vpc_id      = aws_vpc.build_engine_aws_vpc.id
+
+  ingress {
+    description = "Allow SSH"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    description = "Allow k8s API server port"
+    from_port   = 6443
+    to_port     = 6443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    description = "Allow k8s API server port"
+    from_port   = 2379
+    to_port     = 2379
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    description = "Allow UDP connection for longhorn-webhooks"
+    from_port   = 0
+    to_port     = 65535
+    protocol    = "udp"
+    cidr_blocks = ["10.0.0.0/8"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "build_engine_aws_sec_grp_controlplane-${random_string.random_suffix.id}"
+  }
+}
+
+
+# Create worker security group
+resource "aws_security_group" "build_engine_aws_secgrp_worker" {
+  name        = "build_engine_aws_secgrp_worker"
+  description = "Allow all inbound traffic"
+  vpc_id      = aws_vpc.build_engine_aws_vpc.id
+
+  ingress {
+    description = "Allow All Traffic from VPC CIDR block"
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = [aws_vpc.build_engine_aws_vpc.cidr_block]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = {
+    Name = "build_engine_aws_sec_grp_worker-${random_string.random_suffix.id}"
+  }
+}
+
+
+# Create Public subnet
+resource "aws_subnet" "build_engine_aws_public_subnet" {
+  vpc_id     = aws_vpc.build_engine_aws_vpc.id
+  availability_zone = var.aws_availability_zone
+  cidr_block = "10.0.1.0/24"
+
+  tags = {
+    Name = "build_engine_public_subnet-${random_string.random_suffix.id}"
+  }
+}
+
+# Create private subnet
+resource "aws_subnet" "build_engine_aws_private_subnet" {
+  vpc_id     = aws_vpc.build_engine_aws_vpc.id
+  availability_zone = var.aws_availability_zone
+  cidr_block = "10.0.2.0/24"
+
+  tags = {
+    Name = "build_engine_private_subnet-${random_string.random_suffix.id}"
+  }
+}
+
+# Create EIP for NATGW
+resource "aws_eip" "build_engine_aws_eip_nat_gw" {
+  vpc      = true
+
+  tags = {
+    Name = "build_engine_eip_nat_gw-${random_string.random_suffix.id}"
+  }
+}
+
+# Create nat gateway
+resource "aws_nat_gateway" "build_engine_aws_nat_gw" {
```

Comment:
We don't need to create NAT Gateway for this, we only have one node and it has a public IP.

Reply:
I had modified this part of code